src/docs/content/disaster-recovery/setup.md

```diff
@@ -0,0 +1,273 @@
+
+### Perform Autonomous Transaction Processing (ATP) Failover
+
+Go to OCI console and perform a failover.
+``` text
+OCI-Console -> Oracle Database -> Autonomous Transaction Processing (Standby db: ap-hyderabad-1) -> Switchover
+```
+
+{{% alert style="information" icon="information" %}}
+Wait till the switchover completes fully and there are no 'role change in progress' status on either side (Primary and Standby). 
+{{% /alert %}}
+
+### MuShop Setup (Disaster Recovery (DR) site ap-hyderabad-1)
+
+Change your OKE cluster to DR. If you dont have a DR OKE cluster setup yet then refer to the [Create OKE clusters](#create-oke-oracle-cloud-infrastructure-container-engine-for-kubernetes-clusters) section and create a OKE cluster at the DR region.
+
+- Download and extract the DR ADB wallet
+
+```text
+OCI-Console -> Oracle Database -> Autonomous Transaction Processing (Standby db: ap-hyderabad-1) -> DB Connection -> Download wallet
+```
+
+- Create the secrets, set the region as ap-hyderabad-1 in this case
+
+    ```bash
+    kubectl create secret generic oci-credentials \
+    --namespace mushop \
+    --from-literal=tenancy=<TENANCY_OCID> \
+    --from-literal=user=<USER_OCID> \
+    --from-literal=region=<USER_OCI_REGION> \
+    --from-literal=fingerprint=<USER_PUBLIC_API_KEY_FINGERPRINT> \
+    --from-literal=passphrase=<PASSPHRASE_STRING> \
+    --from-file=privatekey=<PATH_OF_USER_PRIVATE_API_KEY>
+    ```
+
+    ```bash
+    kubectl create secret generic oadb-wallet   --namespace mushop   --from-file=/tmp/wallet_remote
+    ```
+    ```bash
+    kubectl create secret generic oadb-admin \
+    --namespace mushop \
+    --from-literal=oadb_admin_pw='DemoAdb#1234'
+    ```
+    ```bash
+    kubectl create secret generic oadb-connection \
+    --namespace mushop \
+    --from-literal=oadb_wallet_pw='DemoAdb#1234' \
+    --from-literal=oadb_service='demoadb_tp'
+    ```
+- Edit/Add the following secrets to values-prod.yaml as shown below
+
+    ```bash
+    cat mushop/values-prod.yaml
+    global:
+    ociAuthSecret: oci-credentials        # OCI authentication credentials secret name
+    ossStreamSecret:                      # Name of Stream Connection secret
+    oadbAdminSecret: oadb-admin           # Name of DB Admin secret created earlier
+    oadbWalletSecret: oadb-wallet         # Name of wallet secret created earlier
+    oadbConnectionSecret: oadb-connection # Name of connection secret created earlier
+    ```
+- Install MuShop
+
+    ```bash
+    helm install -f ./mushop/values-prod.yaml mymushop mushop -n mushop
+    ```
+- Set up the ingress (On DR ap-hyderabad-1)
+
+    ```bash
+    openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout tls.key -out tls.crt -subj "/CN=nginxsvc/O=nginxsvc"
+    ```
+
+    ```bash
+    kubectl create secret tls tls-secret --key tls.key --cert tls.crt -n mushop
+    ```
+
+    ```bash
+    cat << EOF | kubectl -n mushop apply -f -
+    apiVersion: networking.k8s.io/v1beta1
```

Comment:
We use the latest ingress NGINX and the apiVersion is `networking.k8s.io/v1`

Reply:
I picked the example from here https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengsettingupingresscontroller.htm#usingingresscontroller

Reply:
MuShop is ahead of the OKE documentation. We use the latest ingress kubernetes, and need to be `networking.k8s.io/v1`

Reply:
I followed these steps:
cd oci-cloudnative/deploy/complete/helm-chart
helm dependency update ./setup
helm install mushop-utils setup --namespace mushop-utilities --create-namespace
kubectl get svc mushop-utils-ingress-nginx-controller \
--namespace mushop-utilities

However, the ingress fails:

$ cat << EOF | kubectl -n mushop apply -f -
> apiVersion: networking.k8s.io/v1
> kind: Ingress
> metadata:
> name: mushop
> annotations:
>     kubernetes.io/ingress.class: "nginx"
> spec:
> tls:
> - secretName: tls-secret
> rules:
> - http:
>     paths:
>     - backend:
>         serviceName: edge
>         servicePort: 80
> EOF
error: error validating "STDIN": error validating data: [ValidationError(Ingress): unknown field "annotations" in io.k8s.api.networking.v1.Ingress, ValidationError(Ingress): unknown field "name" in io.k8s.api.networking.v1.Ingress, ValidationError(Ingress): unknown field "rules" in io.k8s.api.networking.v1.Ingress, ValidationError(Ingress): unknown field "tls" in io.k8s.api.networking.v1.Ingress]; if you choose to ignore these errors, turn validation off with --validate=false

Reply:
in the setup charts, i see this version:
  - name: ingress-nginx
    version: 3.34.0
    condition: ingress-nginx.enabled
    repository: https://kubernetes.github.io/ingress-nginx

Reply:
@naikvenu We being using `networking.k8s.io/v1 ` for some time without problems. Please check your cluster (version, other ingresses, wrong parameters, etc.). v1beta for this have been deprecated and would be a crime we recommend users to use it.

Check the actual main ingress: https://github.com/oracle-quickstart/oci-cloudnative/blob/a05f2d642d9fcb8052d40554ac025d403e96abc4/deploy/complete/helm-chart/mushop/templates/ingress-edge.yaml#L2

And this is the annotation: https://github.com/oracle-quickstart/oci-cloudnative/blob/a05f2d642d9fcb8052d40554ac025d403e96abc4/deploy/complete/helm-chart/mushop/templates/ingress-edge.yaml#L8-L9

Reply:
Thanks @junior , I'll check